PluralKit.Bot/Commands/Message.cs

```diff
@@ -22,23 +24,69 @@ public class ProxiedMessage
     private static readonly Duration EditTimeout = Duration.FromMinutes(10);
 
     // private readonly IDiscordCache _cache;
+    private readonly ModelRepository _repo;
     private readonly IClock _clock;
+    private readonly IMetrics _metrics;
 
     private readonly EmbedService _embeds;
     private readonly LogChannelService _logChannel;
     private readonly DiscordApiClient _rest;
     private readonly WebhookExecutorService _webhookExecutor;
+    private readonly ProxyService _proxy;
 
     public ProxiedMessage(EmbedService embeds, IClock clock,
-                          DiscordApiClient rest,
+                          DiscordApiClient rest, IMetrics metrics, ModelRepository repo, ProxyService proxy,
                           WebhookExecutorService webhookExecutor, LogChannelService logChannel, IDiscordCache cache)
     {
         _embeds = embeds;
         _clock = clock;
         _rest = rest;
         _webhookExecutor = webhookExecutor;
+        _repo = repo;
         _logChannel = logChannel;
         // _cache = cache;
+        _metrics = metrics;
+        _proxy =  proxy;
+    }
+
+    public async Task ReproxyMessage(Context ctx)
+    {
+        var msg = await GetMessageToEdit(ctx);
```

Comment:
the error messages in this method all mention editing rather than reproxying

Reply:
not sure what's best to do with the errors in `GetMessageToEdit` - passing in a `bool isReproxy` and changing the messages based on that? 🤷

Reply:
yeah, that should be fine